crypto/machine.go

```diff
@@ -180,15 +190,192 @@ func (mach *OlmMachine) HandleToDeviceEvent(evt *event.Event) {
 		switch content := decryptedEvt.Content.Parsed.(type) {
 		case *event.RoomKeyEventContent:
 			mach.receiveRoomKey(decryptedEvt, content)
-			// TODO handle other encrypted to-device events
+		case *event.ForwardedRoomKeyEventContent:
+			if mach.importForwardedRoomKey(decryptedEvt, content) {
+				if ch, ok := mach.roomKeyRequestFilled.Load(content.SessionID); ok {
+					// close channel to notify listener that the key was received
+					close(ch.(chan struct{}))
+				}
+			}
+			// TODO handle m.dummy encrypted to-device event
 		}
-		// TODO handle other unencrypted to-device events. At least m.room_key_request and m.verification.start
+	case *event.RoomKeyRequestEventContent:
+		// Only forward keys to the same user ID
+		if content.Action == event.KeyRequestActionRequest && mach.Client.UserID == evt.Sender {
```

Comment:
This should probably check device verification status somewhere. It can have a flag to disable the check (like `AllowUnverifiedDevices` does for encrypting messages) at least until device verification is actually implemented

Reply:
Done (hardcoded to true for now).